### PR TITLE
Felix htc

### PR DESCRIPTION
**Pull request by felixpacheco**
- Implementation of review comments (except for unit test)
- Implementation of drop specific wells (according to calc.tsv file)
- Implementation of ArgParse in the command line argument parser
- New method that parses the arguments called argument_parser()
- Solve problem with reshape function error 
- Plot according to species and coloraturas according to bioshaker

### CODE REVIEW

File: OD_handler/OD_handler.py [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+import argparse
+
+__version__ = "0.0.1"
+
+
+
+# ------ MAIN RUNNER ------
+
+
+def argument_parser(argv_list=None):
```
Comment (dmccloskey) [inline OD_handler/OD_handler.py]: NIce touch.

File: OD_handler/OD_handler.py [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+    argv_list -- List of arguments provided by the user when running the program.
+    Returned variables:
+    flags
+	'''
+
+	#Initialize the argument parser
+	parser = argparse.ArgumentParser()
+
+	#Adding
+	parser.add_argument("-e", "--estimations", help="Get only the estimations for every sample", action = "store_true")
```
Comment (dmccloskey) [inline OD_handler/OD_handler.py]: I do not remember if these were in there before, but I can say from experience that they greatly improve the user command line experience.

File: OD_handler/OD_handler.py [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+	return flag_all, flag_est, flag_sum, flag_fig, flag_ind, flag_bioshaker
+
+
+
+# ------ PARSE THE DATA USING THE autoflow_parser LIBRARY ------
+
+
+def parse_data() :
+	'''Calls the autoflow_parser and returns a merged xlsx document with all the OD readings combined'''
+	try :
```
Comment (dmccloskey) [inline OD_handler/OD_handler.py]: Very nice and appropriate use of a try except block.  Try except blocks work great for calling sub processes or trying to read external files that may or may not exist.

I am curious though if it is necessary to call the autoflow_parser as a subprocess or could it be called directly from the library?  There are pros and cons to both approaches and I am curious why you have chosen one over the other.

File: OD_handler/OD_handler.py [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+	return df
+
+
+
+# ------ SEPARATION OF GROWTH RATE SAMPLE AND VOLUME LOSS SAMPLES ------
+
+
+def sample_outcome(sample_file, df) : #done
+	'''Uses an external file containing individual sample purposes, returns two classifed dataframes based on sample purposes and labelled by bioshaker.
+	Keyword arguments:
```
Comment (dmccloskey) [inline OD_handler/OD_handler.py]: Thanks for adding the arguments to the docstrings.  This makes it faster to understand the code.

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
+===========
```
Comment (dmccloskey) [inline README.rst]: Documentation is looking good.

**Comment by felixpacheco** [comment]
Shortened __main__.py script
Implementation of plotting options
estimation output as xlsx file

File: README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+# tecan_od_analyzer
+
+
+## Description
+
+
+tecan_od_analyzer is a Python package for analysing optical density (OD) measurements taken from the Tecan Nano plate reader.
```
Comment (dmccloskey) [inline README.md:7]: Nice.

File: tests/test_tecan_od_analyzer.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+ 			self.assertIn(str(name), str(original_IDs))
+
+ 	###IMPLEMENT THE TEST OF THE FLAG
+
+
+
+ 	#----- TEST gr_estimation METHOD -----
+
+ 	
+ 	def test_gr_estimation(self):
```
Comment (dmccloskey) [inline tests/test_tecan_od_analyzer.py:195]: I can see that some of the tests now are looking at the values generated by running the different methods.  Please check that all methods that change the dataframe values have some test in place that checks the correctness of the values.  For example, it looks like there should be something here that tests the different values.

**Pull request merged**